### PR TITLE
Resource Upload + CRUD

### PR DESCRIPTION
**Pull request by T-Dnzt**
# Reason for Change

This pull request adds a way for users to upload resources (only URL type), listing the uploaded resources and managing them (basic CRUD).

# Changes

- Upload Resource: Logged in users can now click on `Upload Resource` in the dropdown menu to access a simple form (title + URL) to create a resource. That resource will be created with the current user as owner, the `resource_type` set to `url` and the privacy set to `priv` (private).
- My Resources: Just like `My Groups`, a user can access a page listing all the resources he has created. He can then view, edit or destroy them. 
- Privacy: A new enum was added to resources (`privacy`). It can either be set to `publ` or `priv` (default to `priv`) and can be used to prevent a resource from being seen by other users. Private resources are not indexed in ElasticSearch. `public` and `private` being reserved keywords in Ruby, I had to use `publ` and `priv` instead.
- Summary Card Changes: I started to improve the summary card design to make them more responsive. I'm planning to continue that effort in a future PR and apply it to all the summary cards. Some adjustments will also be made to some pages to improve the mobile experience.

# Notes

[Related Task](https://trello.com/c/quEqxdJF/13-green-commons-add-a-url-resource-upload-form)

### CODE REVIEW

File: app/views/shared/_nav.html.slim
Comment (ptrikutam) [inline app/views/shared/_nav.html.slim]: Can we change this to say 'Add Resource'? I feel like "Upload" will indicate that we're uploading a file, when in reality we're just linking to a URL.

File: app/views/resources/show.html.slim
Comment (ptrikutam) [inline app/views/resources/show.html.slim]: I'm not sure if this is the best place to keep this, I feel like this should probably be a bit more subtle. Can you add this icon to this location with the text "PRIVATE" next to it? https://cl.ly/1b0i2u2d3e3G

File: spec/models/resource_spec.rb
Comment (ptrikutam) [inline spec/models/resource_spec.rb]: I think the naming here is a bit misleading. Can you update this spec name to be something like `"does not add the #{name} to the search index after creation"`? Or am I mistaken about what the spec is intended to do?

File: spec/models/resource_spec.rb
Comment (ptrikutam) [inline spec/models/resource_spec.rb]: Is this line needed if there is the line below?

File: spec/models/resource_spec.rb
Comment (ptrikutam) [inline spec/models/resource_spec.rb]: Same as above. Can you update this spec name to be `"does not update the #{name} to the search index after update"`?

File: spec/models/resource_spec.rb
Comment (ptrikutam) [inline spec/models/resource_spec.rb]: Same as above. Can you update the spec name?

File: spec/policies/resource_policy_spec.rb
Comment (ptrikutam) [inline spec/policies/resource_policy_spec.rb]: I'm a bit confused here -- what's the difference between the `owner` permissions and the `user` permissions?

In particular, if a user is not the owner of a resource, they shouldn't be able to update / edit / destroy, right?

**Review by ptrikutam — CHANGES_REQUESTED** [review]
This is working & looking great, thanks for putting it together. Requested some minor changes and asked a q or two. Once yo umake those changes I'll approve. Thanks!

File: app/views/resources/show.html.slim
Comment (T-Dnzt) [inline app/views/resources/show.html.slim]: Here's the new version with your recommendations:

private

I think it looks alright like this, I wanted to use a lock instead of the eye but the one provided in Boostrap glyphicon looks terrible. Could add Font Awesome if you think the lock is better than the eye ;)

File: spec/models/resource_spec.rb
Comment (T-Dnzt) [inline spec/models/resource_spec.rb]: Yes, sorry about that. Thanks for spotting it!

File: spec/models/resource_spec.rb
Comment (T-Dnzt) [inline spec/models/resource_spec.rb]: Yup, else I'm getting:

```
#<AddToIndexJob (class)> expected to have received perform_async, but that object is not a spy or method has not been stubbed.
```

We need to setup the stubbing before the expectation.

File: spec/policies/resource_policy_spec.rb
Comment (T-Dnzt) [inline spec/policies/resource_policy_spec.rb]: The specs for `user` are:

```
    it { is_expected.to permit_action(:show) }
    it { is_expected.to permit_action(:index)   }
    it { is_expected.to permit_action(:new)     }
    it { is_expected.to permit_action(:create)  }

    it { is_expected.to forbid_action(:update)  }
    it { is_expected.to forbid_action(:edit)    }
    it { is_expected.to forbid_action(:destroy) }
```

Or can `show`, `index`, `new` and `create` and cannot `update`, `edit` and `destroy` with (`forbid_action`).

For `owner, it's:

```
it { is_expected.to permit_action(:show) }
it { is_expected.to permit_action(:index)   }
it { is_expected.to permit_action(:new)     }
it { is_expected.to permit_action(:create)  }
it { is_expected.to permit_action(:update)  }
it { is_expected.to permit_action(:edit)    }
it { is_expected.to permit_action(:destroy) }
```

Basically, an owner can do everything a user can do plus `update`, `edit` and `destroy`.

**Comment by T-Dnzt** [comment]
@ptrikutam Ready for second review.

File: app/views/resources/show.html.slim
Comment (ptrikutam) [inline app/views/resources/show.html.slim]: I think the lock would be a great idea. Feel free to add font awesome!

File: spec/models/resource_spec.rb
Comment (ptrikutam) [inline spec/models/resource_spec.rb]: Strange -- even though the expectation is to *not* receive the stub. Oh well, works for me.

File: spec/policies/resource_policy_spec.rb
Comment (ptrikutam) [inline spec/policies/resource_policy_spec.rb]: D'oh. I completely misread the text. I didn't read the `forbid_` part of the `forbid_action`. Please disregard my comment :)

**Review by ptrikutam — APPROVED** [review]

**Comment by ptrikutam** [comment]
LGTM 👍  great work!

**Pull request merged**
